datafusion/sql/src/expr/mod.rs

```diff
@@ -350,11 +350,25 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             within_group,
         } = array_agg;
 
-        let order_by = if let Some(order_by) = order_by {
+        let order_by = if let Some(mut order_by) = order_by {
             // TODO: Once sqlparser supports multiple order by clause, handle it
             //       see issue: https://github.com/sqlparser-rs/sqlparser-rs/issues/875
+            let order_by = match order_by.len() {
+                0 => {
+                    return Err(DataFusionError::NotImplemented(
+                        "ARRAY_AGG with empty ORDER BY not supported".to_string(),
+                    ))
+                }
```

Comment:
Hmm isn't `ORDER BY` an optional clause?

For example, in `aggregate_functions.md`:

```
array_agg(expression [ORDER BY expression])
```

Reply:
Oh, nvm, this is the expressions of `ORDER BY`.